core/deployment/src/main/java/io/quarkus/runner/bootstrap/GenerateConfigTask.java

```diff
@@ -0,0 +1,157 @@
+package io.quarkus.runner.bootstrap;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.OpenOption;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.bootstrap.app.CuratedApplication;
+import io.quarkus.builder.BuildChainBuilder;
+import io.quarkus.builder.BuildExecutionBuilder;
+import io.quarkus.builder.BuildResult;
+import io.quarkus.deployment.builditem.ArchiveRootBuildItem;
+import io.quarkus.deployment.builditem.ConfigDescriptionBuildItem;
+
+/**
+ * This phase generates an example configuration file
+ *
+ * @author Stuart Douglas
+ */
+public class GenerateConfigTask {
+
+    private static final Logger log = Logger.getLogger(GenerateConfigTask.class);
+
+    private final Path configFile;
+
+    public GenerateConfigTask(Path configFile) {
+        this.configFile = configFile;
+    }
+
+    public Path run(CuratedApplication application) {
+        //first lets look for some config, as it is not on the current class path
+        //and we need to load it to run the build process
+        try {
+            AugmentActionImpl augmentAction = new AugmentActionImpl(application, Collections.emptyList());
+            BuildResult buildResult = augmentAction.runCustomAction(new Consumer<BuildChainBuilder>() {
+                @Override
+                public void accept(BuildChainBuilder chainBuilder) {
+                    chainBuilder.addFinal(ConfigDescriptionBuildItem.class);
+                    chainBuilder.addInitial(ArchiveRootBuildItem.class);
+                }
+            }, new Consumer<BuildExecutionBuilder>() {
+                @Override
+                public void accept(BuildExecutionBuilder buildExecutionBuilder) {
+                    try {
+                        buildExecutionBuilder.produce(new ArchiveRootBuildItem(Files.createTempDirectory("empty")));
```

Comment:
Shouldn't we `deleteOnExit` this dir? Or do a more controlled cleanup? Perhaps we should consider introducing some kind of Quarkus-post-build-clean-up tasks in perspective.

Reply:
good catch, but in this case it can just be deleted immediately after.